### PR TITLE
Revised Chromosome.jl, Execute.jl, ahwtests.jl to set the active field o...

### PR DESCRIPTION
**Pull request by ahalwright**
...f input and interior nodes during execution instead of in random_chromosome.

### CODE REVIEW

File: src/Chromosome.jl [hunk 1]
```diff
@@ -15,7 +15,9 @@ function Chromosome(p::Parameters)
     interiors = Array(InteriorNode, p.numlevels, p.numperlevel)
     outputs = Array(OutputNode, p.numoutputs)
     fitness = 0.0
-    return Chromosome(p, inputs, interiors, outputs, fitness)
+    active_not_set = false
```
Comment (glesica) [inline src/Chromosome.jl]: This should probably be `active_set` :-)

File: src/Chromosome.jl [hunk 2]
```diff
@@ -59,10 +61,8 @@ function print_chromosome(c::Chromosome, active_only::Bool=false)
     end
 
     for i = 1:length(c.outputs)
-        active = c.outputs[i].active ? "+" : "*"
-        if c.outputs[i].active || !active_only
-            print("[",c.outputs[i].input,"out",i,active,"] ")
-        end
+        active = "+"
```
Comment (glesica) [inline src/Chromosome.jl]: So we print all the nodes as "active" nodes?

File: src/Execute.jl [hunk 3]
```diff
@@ -19,5 +21,37 @@ function evaluate_node(c::Chromosome, node::OutputNode, context::Vector{BitStrin
 end
 
 function execute_chromosome(c::Chromosome, context::Vector{BitString})
+    c.active_set = true
     return BitString[evaluate_node(c, node, context) for node = c.outputs]
 end
+
+# Supplies the standard context for up to 4 inputs
```
Comment (glesica) [inline src/Execute.jl:28]: What is a standard context?

Comment (ahalwright) [inline src/Execute.jl:28]: See the attached notes document.  I think I have a longer version, but I
don't have time to find it now. And I will write up a longer, more
detailed version.

Note the transposed format for the truth table where columns correspond
to rows in a conventional truth table.  We want each output of the
chromosome to be the bit string that represents one of the rows of the
transposed truth table.

For 3 inputs, we want the first input to be 11110000 binary of 0xF0, the
second input to be 11001100 = 0xCC, and the third input to be 10101010 =
0xAA.  So the standard context for 3 inputs would be these three inputs.
 For 4 inputs, they are  0xFF00,  0xF0F0, 0xCCCC, 0xAAAA, and for 2
inputs  0xC, 0xA.

File: src/Chromosome.jl (hunk 2 above)
Comment (ahalwright) [inline src/Chromosome.jl]: All output nodes are automatically active.  But now I don't think that "print_chromosome" should be changing the active status of nodes.

**Pull request merged**